packages/library/README.md

```diff
@@ -452,21 +341,178 @@ function selectShard(method: string): IRpcTransport {
+const transport = createDefaultRpcTransport({ url: 'http:127.0.0.1:8899' });
+
+const rpc = createRpc({ api, transport });
+```
+
+Note that the `createSolanaRpcApi` function is a wrapper on top of the `createRpcApi` function which adds some Solana-specific transformers such as setting a default commitment on all methods or throwing an error when an integer overflow is detected.
+
+#### Creating your Own API Methods
+
+The new library’s RPC specification supports an _infinite_ number of JSON-RPC methods with **zero increase** in bundle size.
+
+This means the library can support future additions to the official [Solana JSON RPC](https://docs.solana.com/api), or [custom RPC methods](https://www.quicknode.com/docs/ethereum/qn_fetchNFTCollectionDetails_v2) defined by some development team – for example, QuickNode or Helius.
+
+Here’s an example of how a developer at QuickNode might build a custom RPC type-spec for their in-house RPC methods:
+
+```tsx
+// Define the method's response payload.
+type NftCollectionDetailsApiResponse = Readonly<{
+    address: string;
+    circulatingSupply: number;
+    description: string;
+    erc721: boolean;
+    erc1155: boolean;
+    genesisBlock: string;
+    genesisTransaction: string;
+    name: string;
+    totalSupply: number;
+}>;
+
+// Set up an interface for the request method.
+interface NftCollectionDetailsApi {
+    // Define the method's name, parameters and response type
+    qn_fetchNFTCollectionDetails(args: { contracts: string[] }): NftCollectionDetailsApiResponse;
+}
+
+// Export the type spec for downstream users.
+export type QuickNodeRpcApi = NftCollectionDetailsApi;
+```
+
+Here’s how a developer might use it:
+
+```tsx
+import { createDefaultRpcTransport, createRpc, createRpcApi } from '@solana/web3.js';
+
+// Create the custom API.
+const api = createRpcApi<QuickNodeRpcApi>();
+
+// Set up an HTTP transport.
+const transport = createDefaultRpcTransport({ url: 'http://127.0.0.1:8899' });
+
+// Create the RPC client.
+const quickNodeRpc = createRpc({ api, transport });
+//    ^ Rpc<QuickNodeRpcApi>
+```
+
+As long as a particular JSON RPC method adheres to the [official JSON RPC specification](https://www.jsonrpc.org/specification), it will be supported by web3.js 2.0.
+
+### Aborting RPC Requests
+
+RPC requests are now abortable with modern `AbortControllers`. When calling an RPC method such as `getSlot`, it will return a `PendingRpcRequest` proxy object that contains a `send` method to send the request to the server.
+
+```ts
+const pendingRequest: PendingRpcRequest<Slot> = rpc.getSlot();
+
+const slot: Slot = await pendingRequest.send();
+```
+
+Since the arguments of the `getSlot` method are reserved for the request payload, the `send` method is the one that can accept additional arguments such as an `AbortSignal` to abort the request.
+
+Here's an example of a custom `AbortController` used to abort a request after a 5-second timeout:
```

Comment:
I'm _slightly_ worried that people will think this is only useful for timeouts, and not – for example – in cases where your UI unmounts, or an operation rolls back. I guess IYKYK.

Reply:
Good shout, I changed the example to something that doesn't use timeout and added another line of documentation:

--- 

[...]

Aborting RPC requests can be useful for a variety of things such as setting a timeout on a request or cancelling a request when a user navigates away from a page.

```tsx
import { createSolanaRpc } from '@solana/web3.js';

const rpc = createSolanaRpc('http://127.0.0.1:8900');

// Create a new AbortController.
const abortController = new AbortController();

// Abort the request when the user navigates away from the current page.
function onUserNavigateAway() {
    abortController.abort();
}

// The request will be aborted if and only if the user navigates away from the page.
const slot = await rpc.getSlot().send({ abortSignal: abortController.signal });
```